storage/innobase/handler/ha_innodb.cc

```diff
@@ -15376,7 +15385,7 @@ ha_innobase::check(
 	}
 
 	/* Restore the original isolation level */
-	m_prebuilt->trx->isolation_level = old_isolation_level;
+	m_prebuilt->trx->isolation_level = old_isolation_level & 3;
```

Comment:
The redundant bitmasking is needed for silencing `-Wconversion` in GCC 6 and later when assigning to a bit-field. Back when I enabled `-Wconversion` in f224525204dcd5a4f907cf9eccbba82ee7ea9018 I checked with https://godbolt.org that it does not incur any extra overhead. Upon reading that commit message, I realize that I’d better test this on the kvm-fulltest2 builder (which is not run for pull requests) to avoid the need of any fixups.